### PR TITLE
[posix-app] fix alarm issue when time changed

### PR DESCRIPTION
**Pull request by bukepo**
Current posix app uses gettimeofday() to get time, which will be affected
by system time changes, and results in alarm not fired in time. This PR uses
clock_gettime() instead to avoid this issue.

### CODE REVIEW

**Comment by abtink** [comment]
Nice find!

**Comment by jwhui** [comment]
Travis is consistently failing on `BUILD_TARGET="toranj-test-framework" VERBOSE=1 TORANJ_POSIX_APP_RCP_MODEL=1`

**Comment by abtink** [comment]
The failing test seems to be the one with largest speedup factor.

**Comment by bukepo** [comment]
I submitted fix.

File: .travis/before_install.sh [hunk 1]
```diff
@@ -64,18 +64,9 @@ cd /tmp || die
     }
 
     [ $BUILD_TARGET != posix-app-pty ] || {
-        sudo apt-get install socat expect libdbus-1-dev autoconf-archive || die
+        sudo apt-get install socat expect || die
         JOBS=$(getconf _NPROCESSORS_ONLN)
         (
-        WPANTUND_TMPDIR=/tmp/wpantund
```
Comment (abtink) [inline .travis/before_install.sh]: Just double-checking. This change here is not related to the main fix, right?

Comment (bukepo) [reply .travis/before_install.sh]: no, not related to the main fix, just by the way.

Comment (abtink) [reply .travis/before_install.sh]: Thanks.

**Comment by smides-nest** [comment]
LGTM
